src/pudl_archiver/archivers/classes.py

```diff
@@ -165,6 +164,42 @@ async def get_hyperlinks(
 
         return hyperlinks
 
+    def validate_archive(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+        resources: dict[str, ResourceInfo],
+    ) -> bool:
+        """Run a series of validation tests for a new archive, and return results.
+
+        Args:
+            baseline_datapackage: DataPackage descriptor from previous version of archive.
+            new_datapackage: DataPackage descriptor from newly generated archive.
+            resources: Dictionary mapping resource name to ResourceInfo.
+
+        Returns:
+            Bool indicating whether or not all tests passed.
+        """
+        validation_tests = self.dataset_validate_archive(
```

Comment:
It seems like this is the results of the tests, rather than the tests themselves. Maybe call it `validation_results`?

Why are there both `validate_archive` and `dataset_validate_archive` methods?

Reply:
Changed to `validation_results`.

As for the two methods, the intention is that `dataset_validate_archive` will be overridden by subclasses to add dataset specific validations. The naming of these two methods is definitely a bit confusing/wonky, but I felt like it would likely change as this evolves, so I didn't worry about it too much. For example, in a separate branch I'm working on, I've replaced `validate_archive` with a method `generate_summary` that returns a `RunSummary` object instead of just a bool.